### PR TITLE
feat: add upgrade-15 (proposal 74)

### PR DESCRIPTION
**Pull request by 0xpatrickdev**
- closes: #152  
- see https://github.com/Agoric/agoric-sdk/tree/dev-upgrade-15/a3p-integration/proposals/a%3Aupgrade-15

### CODE REVIEW

File: proposals/74:upgrade-15/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Proposal to upgrade the chain software to upgrade-14
+
+This software upgrade executes core proposals during the upgrade block, as
+defined by the `agoric-upgrade-14` upgrade handler. See `upgrade14Handler` in
+`agoric-sdk/golang/cosmos/app/app.go`
```
Comment (0xpatrickdev) [inline proposals/74:upgrade-15/README.md:5]:
```suggestion
# Proposal to upgrade the chain software to upgrade-15

This software upgrade executes core proposals during the upgrade block, as
defined by the `agoric-upgrade-15` upgrade handler. See `upgrade15Handler` in
`agoric-sdk/golang/cosmos/app/app.go`
```

**Comment by 0xpatrickdev** [comment]
```
#77 [use-upgrade-15 linux/arm64 internal] load metadata for ghcr.io/agoric/agoric-sdk:unreleased
#77 ERROR: ghcr.io/agoric/agoric-sdk:unreleased: not found
Error: failed to solve: ghcr.io/agoric/agoric-sdk:unreleased: not found
Error: failed with: Error: failed to solve: ghcr.io/agoric/agoric-sdk:unreleased: not found
```
Is this related to #155?

File: proposals/74:upgrade-15/package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "agoricProposal": {
+    "releaseNotes": "https://github.com/Agoric/agoric-sdk/releases/tag/agoric-upgrade-15",
+    "sdkImageTag": "42",
```
Comment (mhofman) [inline proposals/74:upgrade-15/package.json:4]:
```suggestion
    "sdkImageTag": "43",
```

**Review by mhofman — DISMISSED** [review]

File: proposals/74:upgrade-15/exit-reclaim.test.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import test from 'ava';
+import { $ } from 'execa';
+import { execFileSync } from 'node:child_process';
+import { makeAgd, waitForBlock } from './synthetic-chain-excerpt.js';
```
Comment (mhofman) [inline proposals/74:upgrade-15/exit-reclaim.test.js:4]: Use actual `@agoric/synthetic-chain`?

File: proposals/74:upgrade-15/synthetic-chain-excerpt.js
Comment (mhofman) [inline proposals/74:upgrade-15/synthetic-chain-excerpt.js:1]: This file should no longer be necessary.

File: proposals/74:upgrade-15/use.sh
Comment (gibson042) [inline proposals/74:upgrade-15/use.sh:1]: Is it appropriate to _remove_ this file from upgrade-14?

Comment (turadg) [reply proposals/74:upgrade-15/use.sh:1]: I took that out as unrelated to u15

**Comment by anilhelvaci** [comment]
> CI passes! 🎉

What do you think made CI pass? @gibson042 

I'm trying to get `yarn test` build successfully but I get the same error message as the one in CI for https://github.com/Agoric/agoric-3-proposals/pull/157/commits/8a5cb0866e17baeb84f7b2ecf805f566b7682e6c. When I look at the content of https://github.com/Agoric/agoric-3-proposals/pull/157/commits/b2538cea275d6b98c8bf0a696f746d3ae438210d I cannot make sense out of it as to why the changes over there made the CI pass.

What is the purpose of making a bad offer from the upgrade? (see https://github.com/Agoric/agoric-sdk/commit/af9e09d0724ad5e5edb51d79da7fe4414d7f9aea)

**Comment by anilhelvaci** [comment]
> > CI passes! 🎉
> 
> What do you think made CI pass? @gibson042
> 
> I'm trying to get `yarn test` build successfully but I get the same error message as the one in this [CI run](https://github.com/Agoric/agoric-3-proposals/pull/157/checks?check_run_id=25391446982) . When I look at the content of [b2538ce](https://github.com/Agoric/agoric-3-proposals/commit/b2538cea275d6b98c8bf0a696f746d3ae438210d) I cannot make sense out of it as to why the changes over there made the CI pass.
> 
> What is the purpose of making a bad offer from the upgrade? (see [Agoric/agoric-sdk@af9e09d](https://github.com/Agoric/agoric-sdk/commit/af9e09d0724ad5e5edb51d79da7fe4414d7f9aea))

Oh I see that checksums did not match => https://github.com/Agoric/agoric-3-proposals/actions/runs/9228100706/job/25391446982#step:10:4997. However, the latter question still remains;

What is the purpose of making a bad offer from the upgrade? (see [Agoric/agoric-sdk@af9e09d](https://github.com/Agoric/agoric-sdk/commit/af9e09d0724ad5e5edb51d79da7fe4414d7f9aea))

**Comment by turadg** [comment]
> What is the purpose of making a bad offer from the upgrade?

Because the upgrade added a repair for offers in a bad state and it needed such a state in order to confirm the upgrade worked. https://github.com/Agoric/agoric-sdk/issues/9239

**Comment by anilhelvaci** [comment]
Yeah, it makes sense now. Thanks! @turadg

**Pull request merged**